### PR TITLE
Allowing ldap users to view Delivery Services and Servers.

### LINKED ISSUE #493: /api/1.2/deliveryservices.json for LDAP users causes 500 Internal Server error

### PR DESCRIPTION
**Pull request by jheitz200**
Fixes #493

### CODE REVIEW

**Comment by knutsel** [comment]
What happens to the iLO passwords when you are not an admin?

File: traffic_ops/app/lib/UI/Utils.pm [hunk 1]
```diff
@@ -244,6 +244,31 @@ sub is_admin() {
...
+	my $user = $self->current_user()->{username};
+	my $tm_user = $self->db->resultset('TmUser')->search( { username => $user } )->single;
+	if ( defined($tm_user) ) {
+		$ldap_user = undef;
+	}
+	return $ldap_user;
+}
+
+# Returns true if the user has admin or oper privileges or is logged in via ldap.
+sub is_privileged() {
```
Comment (mitchell852) [inline traffic_ops/app/lib/UI/Utils.pm]: are there other places in the code that could leverage this new is_privileged() method? i.e. any other places that check for &is_admin($self) || &is_oper($self)?

**Comment by jheitz200** [comment]
I did not change the logic around iLo passwords, below is how they are being displayed
```
iloPassword: "********"
```

File: traffic_ops/app/lib/API/DeliveryService.pm [hunk 2]
```diff
@@ -36,115 +36,146 @@ my $valid_metric_types = {
...
-	my $self = shift;
-	my $id   = $self->param('id');
 	my @data;
-	my $portal_role = $self->db->resultset('Role')->search( { name => 'portal' } )->get_column('id')->single();
-	my $tm_user    = $self->db->resultset('TmUser')->search( { username => $self->current_user()->{username} } )->single();
-	my $tm_user_id = $tm_user->id;
-	my @ds_ids     = ();
+	if ( defined($rs) ) {
+		while ( my $row = $rs->next ) {
+			next if ( defined($tm_user_id) ) );
```
Comment (mitchell852) [inline traffic_ops/app/lib/API/DeliveryService.pm]: syntax error here per my console: [2016-01-11 10:39:20,051] [ERROR] syntax error at lib/API/DeliveryService.pm line 56, near ") )"

**Pull request merged**